### PR TITLE
Development

### PR DESCRIPTION
**Pull request by Vesper-Works**

### CODE REVIEW

File: OOPAssignment1/Program.cs [hunk 1]
```diff
@@ -11,29 +11,15 @@ class Program
     {
         static void Main()
         {
-            //Local list of integers to hold the first five measurements of the text
-            List<int> parameters = new List<int>();
+            Input input = new Input();
+            input.FileTextInput("example.txt");
```
Comment (Yumeyums) [inline OOPAssignment1/Program.cs:17]: A choice between manual text or pulling from a text file might need to be included, as it only pulls from the example.txt thus far.

File: OOPAssignment1/Analyse.cs [hunk 2]
```diff
@@ -8,29 +8,49 @@ namespace CMP1903M_Assessment_1_Base_Code
...
-            for(int i = 0; i<5; i++)
+
+            //List initialisation
+            for (int i = 0; i < 5; i++)
             {
                 values.Add(0);
             }
 
+            char[] endOfSentance = { '.', '!', '?' }; //Characters which indicate the end of a sentance. Ellipsis will create problems.
+
```
Comment (Yumeyums) [inline OOPAssignment1/Analyse.cs]: Misspelled Sentence, consistency in mispelling however, so it doesn't affect the program when it runs.

File: OOPAssignment1/Analyse.cs [hunk 3]
```diff
@@ -8,29 +8,49 @@ namespace CMP1903M_Assessment_1_Base_Code
...
+        public Dictionary<char, int> LetterFrequency(string input)
+        {
+            Dictionary<char, int> result = new Dictionary<char, int>();
+
+            foreach (char letter in input)
+            {
+                if (!char.IsLetter(letter)) { continue; }
+                result[letter] = result.ContainsKey(letter) ? result[letter] + 1 : 1; //If the dictionary already contains the character, add the character, else increment by one.
+            }
```
Comment (Yumeyums) [inline OOPAssignment1/Analyse.cs:60]: Good comment, since I didn't really understand what the code did, very useful.

**Review by Yumeyums — CHANGES_REQUESTED** [review]
The code is commented, most of it is self-commenting with good naming conventions, as a result it is pretty easy to read and easy to understand. You should've added a choice for the user if they want to enter inputs themselves and while the consistency in the code is great in terms of naming, try not to misspell words. Despite the one I spotted being rather minor, it could impede future code reviews for your other projects.

**Comment by JamesEastwell** [comment]
Overall the code is written exceptionally. The layout, structure and presentation are all very well done with sufficient comments throughout the entire code explaining each segment to a decent standard.
One criticism would be that there seems to be no implementation of the "Manual" and "Text file" option choices. Alongside this, there were minor spelling mistakes but they aren't too impactful, just something to look out for.
Good job.

**Comment by Yumeyums** [comment]
In addition to my previous comments, there is no code that adds a file of "long words" that is created as part of the application, please write this code to meet the full criteria if possible. Overall good work however.

File: OOPAssignment1/Analyse.cs (hunk 2 above)
Comment (Vesper-Works) [reply OOPAssignment1/Analyse.cs]: Good spot, thanks

File: OOPAssignment1/Program.cs (hunk 1 above)
Comment (Vesper-Works) [reply OOPAssignment1/Program.cs:17]: Good idea.

**Pull request merged**